Mixpanel/Mixpanel.m

```diff
@@ -441,20 +448,56 @@ - (void)track:(NSString *)event properties:(NSDictionary *)properties
 #if !MIXPANEL_NO_NOTIFICATION_AB_TEST_SUPPORT
 - (void)setupAutomaticPushTracking {
     SEL selector = nil;
+    Class newCls = [[UNUserNotificationCenter currentNotificationCenter].delegate class];
     Class cls = [[UIApplication sharedApplication].delegate class];
-    if (class_getInstanceMethod(cls, NSSelectorFromString(@"application:didReceiveRemoteNotification:fetchCompletionHandler:"))) {
+
+    if ([[[UIDevice currentDevice] systemVersion] compare:@"10.0" options:NSNumericSearch] != NSOrderedAscending && !newCls) {
+        [[UNUserNotificationCenter currentNotificationCenter] addObserver:self forKeyPath:@"delegate" options:NSKeyValueObservingOptionNew | NSKeyValueObservingOptionOld context:nil];
```

Comment:
i think you can pass 0 for options since it doesn't look like you're using the new/old value below